index/scorch/segment/zap/new.go

```diff
@@ -129,6 +129,54 @@ type interimLoc struct {
 	arrayposs []uint64
 }
 
+// interimDocNums is a wrapper around a roaring Bitmap, and helps as a
+// single roaring.AddRange(0, N) method call is faster than N
+// separate, sequential Add() calls.  This helps with low cardinality
+// fields (e.g., "type" field, or "gender" field), where there are
+// commons runs of sequential docNum's.  The last range of docNum's is
+// tracked as "plus 1" to leverage zero value convention.
+type interimDocNums struct {
+	bs *roaring.Bitmap // may be nil
+
+	lastRangeBegDocNumPlus1 uint64 // inclusive
+	lastRangeEndDocNumPlus1 uint64 // inclusive
+}
+
+func (d *interimDocNums) add(docNum uint64) {
+	docNumPlus1 := docNum + 1
+
+	if d.lastRangeEndDocNumPlus1 != 0 &&
+		d.lastRangeEndDocNumPlus1+1 == docNumPlus1 {
+		d.lastRangeEndDocNumPlus1 = docNumPlus1 // extend the last range
+		return
+	}
+
+	d.incorporateLastRange()
+
+	d.lastRangeBegDocNumPlus1 = docNumPlus1 // start a new last range
+	d.lastRangeEndDocNumPlus1 = docNumPlus1
+}
+
+func (d *interimDocNums) incorporateLastRange() {
+	if d.bs == nil {
+		d.bs = roaring.New()
+	}
+
+	if d.lastRangeBegDocNumPlus1 == 0 {
```

Comment:
does it make sense to have this check and return first at line no: 161

Reply:
> does it make sense to have this check and return first at line no: 161

not in this case.  Even if there's be no docNum's added (or there's no "last range" that's being built up), we still want to end up with a roaring bitmap, even if it's an empty roaring bitmap

Reply:
yea...i was referring to the idea of having a err/fail check to return without doing anything first , followed by a slightly meatier check with some valid initialisation. Not something of any savings..just a cosmetic one.. nm..